### PR TITLE
Cleanup style in InitialSessionState.cs

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Follow #10368 
Fix style issues in InitialSessionState.cs file.

_Please review commit by commit._ There is a lot of commits but everyone is small and simple.

## PR Context

CodeFactor has two nice metrics:
- how many a file was changed
- how many a file has style issues

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 1]
```diff
@@ -203,7 +203,7 @@ public SessionStateTypeEntry(TypeTable typeTable)
         {
             if (typeTable == null)
             {
-                throw PSTraceSource.NewArgumentNullException("typeTable");
+                throw PSTraceSource.NewArgumentException(nameof(typeTable));
```
Comment (vexx32) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: Why the change of exception type? 🙂

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 2]
```diff
@@ -619,8 +619,7 @@ public SessionStateAliasEntry(string name, string definition, string description
         /// <param name="description">A description of the purpose of the alias.</param>
         /// <param name="options">Options defining the scope visibility, readonly and constant.</param>
         /// <param name="visibility"></param>
-        internal SessionStateAliasEntry(string name, string definition, string description,
-            ScopedItemOptions options, SessionStateEntryVisibility visibility)
+        internal SessionStateAliasEntry(string name, string definition, string description, ScopedItemOptions options, SessionStateEntryVisibility visibility)
```
Comment (vexx32) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: These look fairly long... we might be better off putting every parameter on its own line. 🤔

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 3]
```diff
@@ -3480,10 +3488,15 @@ internal void UpdateTypes(ExecutionContext context, bool updateOnly)
...
             // and typically we'll be processing just 2 or 3 files anyway, hence capacity=3.
             ConcurrentDictionary<string, string> filesProcessed
-                = new ConcurrentDictionary<string, string>(/*concurrencyLevel*/3, /*capacity*/3, StringComparer.OrdinalIgnoreCase);
-            Parallel.ForEach(Types, sste =>
-            //            foreach (var sste in Types)
-            {
+                = new ConcurrentDictionary<string, string>(
+                    concurrencyLevel: 3,
+                    capacity: 3,
+                    StringComparer.OrdinalIgnoreCase);
```
Comment (vexx32) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: The line break before `=` doesn't seem necessary. The rest is good though! 👍

**Review by vexx32 — COMMENTED** [review]
Some minor nits and suggestions. Otherwise looks good!

Thanks for taking the time to sort this out! 😊

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 3 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: I will fix in last commit or in follow PR - the file contains still many style issues.

Comment (vexx32) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: Appreciate you taking the time to clean up old code in such an organised manner! 💖 😊

Comment (iSazonov) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: We have already seen that it significantly reduces the time for PR passing through code review.

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 4]
```diff
@@ -3455,9 +3463,15 @@ internal void UpdateTypes(ExecutionContext context, bool updateOnly)
             // Use at most 3 locks (we don't expect contention on that many cores anyways,
             // and typically we'll be processing just 2 or 3 files anyway, hence capacity=3.
             ConcurrentDictionary<string, string> filesProcessed
-                = new ConcurrentDictionary<string, string>(/*concurrencyLevel*/3, /*capacity*/3, StringComparer.OrdinalIgnoreCase);
-            Parallel.ForEach(Types, sste =>
-            {
+                = new ConcurrentDictionary<string, string>(
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: Seems like this line should be part of line 3457.  It was probably split previously because it was long.

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: Should be changed back

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: Done.

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 4 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: Done.

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 5]
```diff
@@ -310,7 +310,7 @@ public SessionStateFormatEntry(FormatTable formattable)
         {
             if (formattable == null)
             {
-                throw PSTraceSource.NewArgumentNullException("formattable");
+                throw PSTraceSource.NewArgumentException(nameof(formattable));
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/InitialSessionState.cs]:
```suggestion
                throw PSTraceSource.NewArgumentNullException(nameof(formattable));
```

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 6]
```diff
@@ -325,7 +325,7 @@ public SessionStateFormatEntry(ExtendedTypeDefinition typeDefinition)
         {
             if (typeDefinition == null)
             {
-                throw PSTraceSource.NewArgumentNullException("typeDefinition");
+                throw PSTraceSource.NewArgumentException(nameof(typeDefinition));
```
Comment (SteveL-MSFT) [inline src/System.Management.Automation/engine/InitialSessionState.cs]:
```suggestion
                throw PSTraceSource.NewArgumentNullException(nameof(typeDefinition));
```

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 2 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: This one seems to be just long enough to be on one line, but ok with either.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]
